bundles/org.openhab.binding.ring/src/main/java/org/openhab/binding/ring/internal/RingVideoServlet.java

```diff
@@ -0,0 +1,124 @@
+import java.nio.file.FileSystem;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.jetty.http.HttpMethod;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.http.HttpService;
+import org.osgi.service.http.NamespaceException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Main OSGi service and HTTP servlet for Ring Video
+ *
+ * @author Peter Mietlowski (zolakk) - Initial contribution
+ * @author Ben Rosenblum - Updated for OH4 / New Maintainer
+ */
+@Component(service = HttpServlet.class)
+@NonNullByDefault
+public class RingVideoServlet extends HttpServlet {
+
+    private static final long serialVersionUID = -5592161948589682812L;
+
+    private final Logger logger = LoggerFactory.getLogger(RingVideoServlet.class);
+
+    private String videoStoragePath = "";
+    /**
+     * The RestClient is used to connect to the Ring Account.
+     */
+    private @Nullable RestClient restClient;
+
+    public RingVideoServlet() {
+    }
+
+    public RingVideoServlet(HttpService httpService, String videoStoragePath) {
+        Path path = Paths.get(videoStoragePath);
+        FileSystem fs = path.getFileSystem();
+        String sep = fs.getSeparator();
+        this.videoStoragePath = videoStoragePath + (videoStoragePath.endsWith(sep) ? "" : sep);
+        restClient = new RestClient();
+        try {
+            httpService.registerServlet(SERVLET_VIDEO_PATH, this, null, httpService.createDefaultHttpContext());
+        } catch (NamespaceException | ServletException e) {
+            logger.warn("Register servlet fails", e);
+        }
+    }
+
+    @SuppressWarnings("null")
+    @Override
+    protected void service(HttpServletRequest request, HttpServletResponse response)
+            throws ServletException, IOException {
+        try {
+            String ipAddress = request.getHeader("HTTP_X_FORWARDED_FOR");
+            if (ipAddress == null) {
+                ipAddress = request.getRemoteAddr();
+            }
+            String path = request.getRequestURI().substring(0, SERVLET_VIDEO_PATH.length());
+            logger.trace("RingVideo: Request from {}:{}{} ({}:{}, {})", ipAddress, request.getRemotePort(), path,
+                    request.getRemoteHost(), request.getServerPort(), request.getProtocol());
+            if (!request.getMethod().equalsIgnoreCase(HttpMethod.GET.toString())) {
+                logger.error("RingVideo: Unexpected method='{}'", request.getMethod());
+            }
+            if (!path.equalsIgnoreCase(SERVLET_VIDEO_PATH)) {
+                logger.error("RingVideo: Invalid request received - path = {}", path);
```

Comment:
do not use `ERROR` (see other comment), also below

Reply:
Fixed.